### PR TITLE
ANTIC-026 Profile data gets displayed and is editable

### PR DESCRIPTION
**Pull request by naggab**
Profile information is displayed. 

Now the profile information can be changed and stored. To do so change your field (Email is not editable) and press the Edit button

### CODE REVIEW

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt [hunk 1]
```diff
@@ -5,30 +5,112 @@ import android.view.LayoutInflater
...
+                val user = User(
+                    oldUser.id,
+                    editTextName.text.toString(),
+                    editTextSurname.text.toString(),
+                    oldUser.email,
+                    editTextAddress.text.toString(),
+                    editTextSocialSecurityID.text.toString(),
+                    editTextPhoneNumber.text.toString(),
+                    oldUser.password
+                )
```
Comment (DinosaurierRex) [inline AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt:79]: Please check some of the input fields that they aren't empty (e.g. see Register Fragment)

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt [hunk 2]
```diff
@@ -5,30 +5,112 @@ import android.view.LayoutInflater
...
+
+                    override fun onFailure() {
+                        btnedit.isEnabled = true
+                        Toast.makeText(
+                            requireContext(),
+                            getString(R.string.error_firebase_communication),
+                            Toast.LENGTH_LONG
+                        ).show()
+                    }
+                })
```
Comment (DinosaurierRex) [inline AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt:99]: Please add some notification to the user so that he knows whats going on (fail, success ...)

**Review by DinosaurierRex — CHANGES_REQUESTED** [review]

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt (hunk 1 above)
Comment (naggab) [reply AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt:89]: Fixed with our new commit

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt (hunk 2 above)
Comment (naggab) [reply AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt]: Fixed with our new commit

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt (hunk 1 above)
Comment (DinosaurierRex) [reply AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt:89]: LGTM 👍

**Review by DinosaurierRex — APPROVED** [review]
LGTM 👍

File: AntiCovid/app/src/androidTest/java/com/swtug/anticovid/startUp/RegisterTest.kt [hunk 3]
```diff
@@ -45,7 +44,7 @@ class RegisterTest {
         }
     }
 
-    private fun fillOutRegisterForm(email: String, pwd: String, secID: String) {
+    private fun fillOutRegisterForm(email: String, pwd: String) {
```
Comment (DinosaurierRex) [inline AntiCovid/app/src/androidTest/java/com/swtug/anticovid/startUp/RegisterTest.kt:47]: Please revert these changes. Since you actually provided a newer version already for the whole test

File: AntiCovid/app/src/androidTest/java/com/swtug/anticovid/startUp/RegisterTest.kt [hunk 4]
```diff
@@ -96,31 +95,9 @@ class RegisterTest {
...
-        fillOutRegisterForm(TEST_USER_MAIL, TEST_USER_PWD, "12345678910")
-        Thread.sleep(1000)
-
-        onView(withId(R.id.buttonSignUp)).perform(click())
-
-        val compareString = TestUtils.getResoureceString(getContext(), R.string.error_secID_lenth)
-        onView(withId(R.id.editTextInputError)).check(matches(withText(compareString)))
-    }
-
     @Test
```
Comment (DinosaurierRex) [inline AntiCovid/app/src/androidTest/java/com/swtug/anticovid/startUp/RegisterTest.kt:98]: same as above for the whole test

**Review by DinosaurierRex — CHANGES_REQUESTED** [review]
Please see comments

File: AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt [hunk 5]
```diff
@@ -65,10 +108,81 @@ class ProfileFragment : BaseFragment() {
...
+            FirebaseRepo.updateUser(user, object : FirebaseListener {
+                override fun onSuccess(user: User?) {
+                    btnedit.isEnabled = true
+                    Toast.makeText(
+                        requireContext(),
+                        requireContext().getString(R.string.changes_saved),
+                        Toast.LENGTH_LONG
+                    ).show()
+                }
+
```
Comment (DinosaurierRex) [inline AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt:169]: It would make sense to only save the User in our cache, if Firebase returns a SUCCESS. Also a revert to old user data would then be cool to actually always show the correct data which comes from Firebase

**Review by DinosaurierRex — CHANGES_REQUESTED** [review]
